### DISCUSSION TITLE
kepler in container

### DISCUSSION THREAD

**Discussion by jochen-schuettler** [sustainable-computing-io/kepler#1084] [Q&A] [answered]
As I understood from the documentation there is fundamental difference between running Kepler on bare-metal or in a VM, obviously. The latter feeds from a model or from kepler also running on the bare-metal machine, feeding the VM instance. But I'm not sure about kepler in a container (on bare-metal), is that like bare-metal use or like VM use? Or, what host path need to be mounted into the container for bare-metal use, if so?

**Comment by rootfs** [comment]
kepler container is measured the same way on both setups. The `/sys` path is what is needed to find the container ID and container names

**Reply by jochen-schuettler** [reply]
Do I understand correctly that kepler in container is behaving (measuring) like kepler in VM?

**Reply by rootfs** [reply]
the energy estimate algorithms and metrics that are available from kernel are different but otherwise kepler is doing the same

**Reply by marceloamaral** [reply, marked as answer]
It doesn't matter much whether Kepler is running in a container; what matters is whether Kepler is running on bare-metal or VMs.
That is, if hardware power meters are available or not.

**Comment by jochen-schuettler** [comment]
I find [this answer](https://github.com/sustainable-computing-io/kepler/discussions/1084#discussioncomment-8135345) is the most concise and helpful one.
